File: src/main/java/net/volcano/jdacommands/model/menu/pagers/EmbedEntirePagerBuilder.java
package net.volcano.jdacommands.model.menu.pagers;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.EmbedType;
import net.dv8tion.jda.api.entities.MessageEmbed;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class EmbedEntirePagerBuilder extends EmbedPagerBuilder {
	
	private List<MessageEmbed> embeds = new ArrayList<>();
	
	@Override
	protected EmbedPager buildEmbed(MessageEmbed baseEmbed) {
		return new EmbedEntirePager(embeds, getExpiration());
	}
	
	public void setEmbeds(List<EmbedBuilder> embeds) {
		this.embeds = embeds.stream().map(EmbedBuilder::build).collect(Collectors.toList());
	}
	
	public void addEmbed(EmbedBuilder embed) {
		embeds.add(embed.build());
	}
	
	public void next() {
		embeds.add(new MessageEmbed(
				getUrl(),
				getTitle(),
				getDescription().toString(),
				EmbedType.RICH,
				getTimestamp(),
				getColor(),
				getThumbnail(),
				null,
				getAuthor(),
				null,
				getFooter(),
				getImage(),
				getFields()
		));
		clear();
	}
	
}

File: src/main/java/net/volcano/jdacommands/model/menu/pagers/EmbedEntirePagerBuilder.kt
package net.volcano.jdacommands.model.menu.pagers

import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.entities.EmbedType
import net.dv8tion.jda.api.entities.MessageEmbed
import java.util.*

class EmbedEntirePagerBuilder : EmbedPagerBuilder() {

	private val embeds = mutableListOf<MessageEmbed>()

	override fun buildEmbed(baseEmbed: MessageEmbed): EmbedPager {
		next()
		return EmbedEntirePager(embeds, expiration)
	}

	fun setEmbeds(embeds: List<EmbedBuilder>) {
		this.embeds.clear()
		this.embeds.addAll(embeds.map { it.build() })
	}

	fun addEmbed(embed: EmbedBuilder) {
		this.embeds.add(embed.build())
	}

	fun next() {

		embeds.add(
			MessageEmbed(
				url,
				title,
				description.toString(),
				EmbedType.RICH,
				timestamp,
				color,
				thumbnail,
				null,
				author,
				null,
				footer,
				image,
				LinkedList(fields)
			)
		)

		this.clear()

	}

}